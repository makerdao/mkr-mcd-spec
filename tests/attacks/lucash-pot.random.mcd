STEPS ( ATTACK-PRELUDE )

AddGenerator ( GenPotFileDSR
             ; GenTimeStep
             ; GenPotJoin "Alice"
             ; GenPotDrip
             ; GenPotExit "Alice"
             )

GenSteps
