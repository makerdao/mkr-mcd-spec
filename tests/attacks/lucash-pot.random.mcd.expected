<kmcd-random>
  <kmcd-properties>
    <kmcd>
      <kmcd-driver>
        <k>
          .
        </k>
        <return-value>
          .
        </return-value>
        <msg-sender>
          "Alice"
        </msg-sender>
        <this>
          "Alice"
        </this>
        <current-time>
          2
        </current-time>
        <call-stack>
          .List
        </call-stack>
        <pre-state>
          .
        </pre-state>
        <events>
          .List
        </events>
        <frame-events>
          .List
        </frame-events>
      </kmcd-driver>
      <kmcd-state>
        <cat>
          <cat-wards>
            SetItem ( End )
          </cat-wards>
          <cat-ilks>
            .Map
          </cat-ilks>
          <cat-live>
            true
          </cat-live>
          <cat-vow>
            0
          </cat-vow>
        </cat>
        <dai>
          <dai-wards>
            .Set
          </dai-wards>
          <dai-totalSupply>
            0
          </dai-totalSupply>
          <dai-account-id>
            0
          </dai-account-id>
          <dai-balance>
            .Map
          </dai-balance>
          <dai-allowance>
            .Map
          </dai-allowance>
          <dai-nonce>
            .Map
          </dai-nonce>
        </dai>
        <end-state>
          <endPhase>
            false
          </endPhase>
          <end>
            <end-wards>
              .Set
            </end-wards>
            <end-live>
              true
            </end-live>
            <end-when>
              0
            </end-when>
            <end-wait>
              0
            </end-wait>
            <end-debt>
              0
            </end-debt>
            <end-tag>
              .Map
            </end-tag>
            <end-gap>
              "gold" |-> 0
            </end-gap>
            <end-art>
              .Map
            </end-art>
            <end-fix>
              .Map
            </end-fix>
            <end-bag>
              "Alice" |-> 0
              "Bobby" |-> 0
            </end-bag>
            <end-out>
              { "gold" , "Alice" } |-> 0
              { "gold" , "Bobby" } |-> 0
            </end-out>
          </end>
        </end-state>
        <flap-state>
          <flap-wards>
            SetItem ( Vow )
          </flap-wards>
          <flap-bids>
            .Map
          </flap-bids>
          <flap-kicks>
            0
          </flap-kicks>
          <flap-live>
            true
          </flap-live>
          <flap-beg>
            21 /Rat 20
          </flap-beg>
          <flap-ttl>
            10800
          </flap-ttl>
          <flap-tau>
            172800
          </flap-tau>
        </flap-state>
        <flips>
          <flip>
            <flip-ilk>
              "gold"
            </flip-ilk>
            <flip-wards>
              SetItem ( End )
            </flip-wards>
            <flip-bids>
              .Map
            </flip-bids>
            <flip-beg>
              21 /Rat 20
            </flip-beg>
            <flip-ttl>
              10800
            </flip-ttl>
            <flip-tau>
              172800
            </flip-tau>
            <flip-kicks>
              0
            </flip-kicks>
          </flip>
        </flips>
        <flop-state>
          <flop-wards>
            SetItem ( Vow )
          </flop-wards>
          <flop-bids>
            .Map
          </flop-bids>
          <flop-kicks>
            0
          </flop-kicks>
          <flop-live>
            true
          </flop-live>
          <flop-beg>
            21 /Rat 20
          </flop-beg>
          <flop-pad>
            3 /Rat 2
          </flop-pad>
          <flop-ttl>
            10800
          </flop-ttl>
          <flop-tau>
            172800
          </flop-tau>
          <flop-vow>
            0
          </flop-vow>
        </flop-state>
        <gems>
          <gem>
            <gem-id>
              "MKR"
            </gem-id>
            <gem-wards>
              .Set
            </gem-wards>
            <gem-balances>
              "Alice" |-> 0
              "Bobby" |-> 0
              Flap |-> 20
              GemJoin "MKR" |-> 0
              Vow |-> 0
            </gem-balances>
          </gem> <gem>
            <gem-id>
              "gold"
            </gem-id>
            <gem-wards>
              .Set
            </gem-wards>
            <gem-balances>
              "Alice" |-> 10
              "Bobby" |-> 10
              GemJoin "gold" |-> 20
            </gem-balances>
          </gem>
        </gems>
        <join-state>
          <gem-joins>
            <gem-join>
              <gem-join-gem>
                "MKR"
              </gem-join-gem>
              <gem-join-wards>
                .Set
              </gem-join-wards>
              <gem-join-live>
                true
              </gem-join-live>
            </gem-join> <gem-join>
              <gem-join-gem>
                "gold"
              </gem-join-gem>
              <gem-join-wards>
                .Set
              </gem-join-wards>
              <gem-join-live>
                true
              </gem-join-live>
            </gem-join>
          </gem-joins>
          <dai-join>
            <dai-join-wards>
              .Set
            </dai-join-wards>
            <dai-join-live>
              true
            </dai-join-live>
          </dai-join>
        </join-state>
        <jug>
          <jug-wards>
            .Set
          </jug-wards>
          <jug-ilks>
            .Map
          </jug-ilks>
          <jug-vow>
            0
          </jug-vow>
          <jug-base>
            0
          </jug-base>
        </jug>
        <pot>
          <pot-wards>
            SetItem ( End )
          </pot-wards>
          <pot-pies>
            "Alice" |-> 0
            "Bobby" |-> 0
          </pot-pies>
          <pot-pie>
            0
          </pot-pie>
          <pot-dsr>
            597 /Rat 500
          </pot-dsr>
          <pot-chi>
            356409 /Rat 250000
          </pot-chi>
          <pot-vow>
            Vow
          </pot-vow>
          <pot-rho>
            2
          </pot-rho>
          <pot-live>
            true
          </pot-live>
        </pot>
        <spot>
          <spot-wards>
            .Set
          </spot-wards>
          <spot-ilks>
            "gold" |-> SpotIlk ( 1 , 1 )
          </spot-ilks>
          <spot-par>
            1
          </spot-par>
          <spot-live>
            true
          </spot-live>
        </spot>
        <vat>
          <vat-wards>
            SetItem ( End )
            SetItem ( GemJoin "gold" )
            SetItem ( Pot )
          </vat-wards>
          <vat-can>
            "Alice" |-> SetItem ( End )
            SetItem ( Flip "gold" )
            SetItem ( Pot )
            "Bobby" |-> SetItem ( End )
            SetItem ( Flip "gold" )
            SetItem ( Pot )
            End |-> .Set
            Flap |-> .Set
            Pot |-> .Set
            Vow |-> .Set
          </vat-can>
          <vat-ilks>
            "gold" |-> Ilk ( 20 , 1 , 3000000000 , 1000000000000 , 0 )
          </vat-ilks>
          <vat-urns>
            { "gold" , "Alice" } |-> Urn ( 10 , 10 )
            { "gold" , "Bobby" } |-> Urn ( 10 , 10 )
            { "gold" , End } |-> Urn ( 0 , 0 )
          </vat-urns>
          <vat-gem>
            { "gold" , "Alice" } |-> 0
            { "gold" , "Bobby" } |-> 0
            { "gold" , End } |-> 0
            { "gold" , Flip "gold" } |-> 0
          </vat-gem>
          <vat-dai>
            "Alice" |-> 10
            "Bobby" |-> 10
            End |-> 0
            Flap |-> 0
            Pot |-> 0
            Vow |-> 0
          </vat-dai>
          <vat-sin>
            "Alice" |-> 0
            "Bobby" |-> 0
            End |-> 0
            Flap |-> 0
            Pot |-> 0
            Vow |-> 0
          </vat-sin>
          <vat-debt>
            20
          </vat-debt>
          <vat-vice>
            0
          </vat-vice>
          <vat-Line>
            1000000000000
          </vat-Line>
          <vat-live>
            true
          </vat-live>
        </vat>
        <vow>
          <vow-wards>
            SetItem ( End )
            SetItem ( Pot )
          </vow-wards>
          <vow-sins>
            .Map
          </vow-sins>
          <vow-sin>
            0
          </vow-sin>
          <vow-ash>
            0
          </vow-ash>
          <vow-wait>
            0
          </vow-wait>
          <vow-dump>
            0
          </vow-dump>
          <vow-sump>
            0
          </vow-sump>
          <vow-bump>
            1000000000
          </vow-bump>
          <vow-hump>
            0
          </vow-hump>
          <vow-live>
            true
          </vow-live>
        </vow>
      </kmcd-state>
    </kmcd>
    <processed-events>
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . rely Pot ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . rely End ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Pot . rely End ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vow . rely Pot ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vow . rely End ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Cat . rely End ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Flap . rely Vow ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Flop . rely Vow ) )
      ListItem ( Measure ( 0 , .Map , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser Vow ) )
      ListItem ( Measure ( 0 , Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser Pot ) )
      ListItem ( Measure ( 0 , Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser Flap ) )
      ListItem ( Measure ( 0 , Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser End ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Pot . file vow-file Vow ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "gold" . init ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , GemJoin "gold" . init ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . rely GemJoin "gold" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initGem "gold" End ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initGem "gold" Flip "gold" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initCDP "gold" End ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Spot . init "gold" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Spot . setPrice "gold" 1 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Flip "gold" . init ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Flip "gold" . rely End ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , End . initGap "gold" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . init ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , GemJoin "MKR" . init ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . initUser Vow ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . initUser Flap ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . mint Flap 20 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . file Line 1000000000000 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initIlk "gold" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . file spot "gold" 3000000000 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . file line "gold" 1000000000000 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vow . file bump 1000000000 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "gold" . initUser "Alice" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "gold" . initUser "Bobby" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "gold" . mint "Alice" 20 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "gold" . mint "Bobby" 20 ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . initUser "Alice" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Gem "MKR" . initUser "Bobby" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Pot . initUser "Alice" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Pot . initUser "Bobby" ) )
      ListItem ( Measure ( 0 , End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser "Alice" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initGem "gold" "Alice" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initCDP "gold" "Alice" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initUser "Bobby" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initGem "gold" "Bobby" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Vat . initCDP "gold" "Bobby" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . hope Pot ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . hope Flip "gold" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . hope End ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Bobby" , Vat . hope Pot ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Bobby" , Vat . hope Flip "gold" ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Bobby" , Vat . hope End ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . slip "gold" "Alice" 10 ) )
      ListItem ( LogNote ( "Alice" , Gem "gold" . transferFrom "Alice" GemJoin "gold" 10 ) )
      ListItem ( LogNote ( "Alice" , GemJoin "gold" . join "Alice" 10 ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Bobby" , Vat . slip "gold" "Bobby" 10 ) )
      ListItem ( LogNote ( "Bobby" , Gem "gold" . transferFrom "Bobby" GemJoin "gold" 10 ) )
      ListItem ( LogNote ( "Bobby" , GemJoin "gold" . join "Bobby" 10 ) )
      ListItem ( Measure ( 0 , "Alice" |-> 0
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 0 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . frob "gold" "Alice" "Alice" "Alice" 10 10 ) )
      ListItem ( Measure ( 10 , "Alice" |-> 10
      "Bobby" |-> 0
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 10 , 0 , 0 ) )
      ListItem ( LogNote ( "Bobby" , Vat . frob "gold" "Bobby" "Bobby" "Bobby" 10 10 ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , End . initBag "Alice" ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , End . initBag "Bobby" ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , End . initOut "gold" "Alice" ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , End . initOut "gold" "Bobby" ) )
      ListItem ( GenStep ( b"aa" , transact ADMIN Pot . file dsr 597 /Rat 500 ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ADMIN , Pot . file dsr 597 /Rat 500 ) )
      ListItem ( GenStep ( b"aa" , TimeStep 2 ) )
      ListItem ( TimeStep ( 2 , 2 ) )
      ListItem ( GenStep ( b"aa" , transact "Alice" Pot . join 97 /Rat 10 ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( Exception ( Pot . join 97 /Rat 10 ) )
      ListItem ( GenStep ( b"a" , transact ANYONE Pot . drip ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 1 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( ANYONE , Vat . suck Vow Pot 0 ) )
      ListItem ( LogNote ( ANYONE , Pot . drip ) )
      ListItem ( GenStep ( b"a" , transact "Alice" Pot . exit 0 ) )
      ListItem ( Measure ( 20 , "Alice" |-> 10
      "Bobby" |-> 10
      End |-> 0
      Flap |-> 0
      Pot |-> 0
      Vow |-> 0 , 356409 /Rat 250000 , 0 , 20 , 0 , 0 ) )
      ListItem ( LogNote ( "Alice" , Vat . move Pot "Alice" 0 ) )
      ListItem ( LogNote ( "Alice" , Pot . exit 0 ) )
    </processed-events>
    <properties>
      "Debt Constant After Thaw" |-> debtConstantAfterThaw
      "Pot Interest Accumulation After End" |-> potEndInterest
      "PotChi PotPie VatPot" |-> potChiPieDai
      "Total Backed Debt Consistency" |-> totalBackedDebtConsistency
      "Total Bound on Debt" |-> totalDebtBoundedRun ( 20 , 597 /Rat 500 )
      "Unauthorized Flap Kick" |-> unAuthFlapKick
      "Unauthorized Flip Kick" |-> unAuthFlipKick
      "Zero-Time Pot Interest Accumulation" |-> zeroTimePotInterest
    </properties>
  </kmcd-properties>
  <kmcd-gen>
    <random>
      b""
    </random>
    <used-random>
      b""
    </used-random>
    <generator-next>
      1
    </generator-next>
    <generator-current>
      0
    </generator-current>
    <generator-remainder>
      .GenStep
    </generator-remainder>
    <generators>
      <generator>
        <generator-id>
          0
        </generator-id>
        <generator-steps>
          .GenStep
        </generator-steps>
      </generator>
    </generators>
  </kmcd-gen>
</kmcd-random>
